src/MainPage.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div id="app">
+    <BaseLayout>
+      <template v-slot:header>
+        <Header>
+          <div class="row">
+            <div class="col">
+              <AppNameLabel />
+            </div>
+          </div>
+          <div class="row">
+            <div class="col text-white">
+              <h1>FIND YOUR MOVIE</h1>
+            </div>
+          </div>
+          <div class="row">
+            <div class="col">
+              <SearchBar
+                :firstOption="searchByFirstOption"
+                :secondOption="searchBySecondOption"
+                @button-clicked="onSearchBarButtonClicked"
+              />
+            </div>
+          </div>
+        </Header>
+      </template>
+
+      <template v-slot:toolbar>
+        <div class="row align-items-center h-100">
+          <div class="col">
+            <span class="text-white pl-5">
+              {{ movies.length }} movie found
+            </span>
+          </div>
+          <div class="col">
+            <Toogle
+              class="float-right"
+              :label="sortByLabel"
+              :firstOption="sortByFirstOption"
+              :secondOption="sortBySecondOption"
+              @value-switched="onSortByToogleValueSwitched"
+            />
+          </div>
+        </div>
+      </template>
+
+      <FilmCardGallery
+        class="gallery__bg-dark p-4"
+        :cards="movies"
+        @card-clicked="onFilmCardClick"
+      />
+    </BaseLayout>
+  </div>
+</template>
+
+<script>
+import AppNameLabel from "./components/AppNameLabel";
+import Header from "./components/Header";
+import SearchBar from "./components/SearchBar";
+import FilmCardGallery from "./components/FilmCardGallery";
+import Toogle from "./components/Toogle";
+import BaseLayout from "./components/layout/BaseLayout";
+
+import { publicPath } from "../vue.config";
+import { appConfig } from "./config/appConfig";
+import { mockData } from "./config/mockData";
+
+export default {
+  name: "MainPage",
+  components: {
+    BaseLayout,
+    AppNameLabel,
+    Header,
+    SearchBar,
+    Toogle,
+    FilmCardGallery
+  },
+  data: () => {
+    return {
+      sortByLabel: appConfig.sortConfig.label,
+      sortByFirstOption: appConfig.sortConfig.firstOption,
+      sortBySecondOption: appConfig.sortConfig.secondOption,
+
+      searchByFirstOption: appConfig.searchConfig.firstOption,
+      searchBySecondOption: appConfig.searchConfig.secondOption,
+
+      movies: mockData.movies
+    };
+  },
+  methods: {
+    onSortByToogleValueSwitched(sortByValue) {
+      console.log("onSortByToogleValueSwitched", sortByValue);
+      if (this.sortByFirstOption === sortByValue) {
+        this.movies.sort(
```

Comment:
Interesting approach.
However, there's an implicit reassignment on every click of the sorting toggle.
Consider making 2 computed properties and do the sorting in there.
Also, copy original object before sorting to avoid mutation of original array.

Reply:
I don't quite understand what you mean. What 2 computed properties should it be? But I tried a new solution.

1. replace **movies** prop with **new computed properties**:
https://github.com/skrypetsviacheslav/vue-mentoring/blob/464bd2e614a06a0f47cc160186a5f71218162106/src/MainPage.vue#L47-L51
https://github.com/skrypetsviacheslav/vue-mentoring/blob/464bd2e614a06a0f47cc160186a5f71218162106/src/MainPage.vue#L94-L104
2. add prop that store selected option:
https://github.com/skrypetsviacheslav/vue-mentoring/blob/464bd2e614a06a0f47cc160186a5f71218162106/src/MainPage.vue#L85
3. change new prop on value-switched event:
https://github.com/skrypetsviacheslav/vue-mentoring/blob/464bd2e614a06a0f47cc160186a5f71218162106/src/MainPage.vue#L107-L110